packages/checkout-storefront/src/components/SelectBox/SelectBox.tsx

```diff
@@ -1,31 +1,42 @@
 import clsx from "clsx";
 import { Children, Classes } from "@/checkout-storefront/lib/globalTypes";
-import { FC, HTMLAttributes } from "react";
+import { HTMLAttributes } from "react";
+import { useFormContext } from "@/checkout-storefront/hooks/useForm";
+import { useField } from "formik";
 
-export interface SelectBoxProps
+export interface SelectBoxProps<TFieldName extends string>
   extends Classes,
     Children,
     Omit<HTMLAttributes<HTMLInputElement>, "children"> {
-  value: string;
-  selectedValue: string | undefined;
   disabled?: boolean;
+  name: TFieldName;
+  value: string;
 }
 
-export const SelectBox: FC<SelectBoxProps> = ({
+export const SelectBox = <TFieldName extends string>({
   children,
-  value,
-  selectedValue,
   className,
-  id,
   disabled = false,
-  ...rest
-}) => {
-  const selected = selectedValue === value;
+  name,
+  value,
+  id,
+}: SelectBoxProps<TFieldName>) => {
+  const identifier = id || value;
```

Comment:
This is a solution for us having in some cases two forms in checkout with same named fields (e.g billing and shipping address forms), hence without separate id's html goes crazy when making changes. So in case of stuff like delivery methods (which are always only one per page), the box relies on the basic flow with value. And when we have stuff like billing and shipping list selection, we deliberately give them unique ids, so we can easily keep the simple form data flow in their corresponding forms (which also use the same hooks and types underneath), while html working fine.

Reply:
ok, makes sense this makes reading the data from form easier 👍🏻 
plz write a comment though because it felt really weird when I saw it 😅

Reply:
Will also add commend in the code to indicate that

Reply:
ping @bmigirl I don't see the changes